### PR TITLE
🇪🇸 cwltool compatibility

### PR DESCRIPTION
**Pull request by dmiller15**
## Description

These are changes necessary to allow our alignment workflow to run end-to-end using cwltool:
- Pulled out the rg_string value from the argument block. When left in the argument block, cwltool would not "interpret" the value. So if you hand it ('hello\\tworld') it would explicitly pass the double escaped tab ('\\t') rather than simply a tab ('\t'). The solution was to split the argument block into things that come before and after it then use inputBinding to put it between them.
- No longer rm bams in gatherbamfiles. The bams it tries to delete are typically readonly files in cwltool and users won't have rm permissions.
- Rewrote the flatten function in sambamba merge anylist. The old function was only flattening a single level so I rewrote it to recursively flatten. All other changes are just cleaning up the code in that tool.

Resolves https://github.com/d3b-center/bixu-tracker/issues/591

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Ran the tool end to end using cwltool on my personal VM
- [x] IPC folks ran it as well and were satisfied with the results

**Test Configuration**:
* Environment: cwltool 3.0
* Test files: https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/files/5ec6a6c3e4b0e4c51669012b/, https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/files/5ecd251be4b0e4c5b1fdfff1/, https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/files/5ecd251be4b0e4c5b1fdfff2/

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: tools/bwa_mem_naive.cwl [hunk 1]
```diff
@@ -25,21 +25,23 @@ arguments:
...
+    valueFrom: >-
       ${return inputs.reads.path}
       ${if (inputs.mates != null) {return inputs.mates.path} else {return ""}}
       | /opt/samblaster/samblaster -i /dev/stdin -o /dev/stdout
       | /opt/sambamba_0.6.3/sambamba_v0.6.3 view -t 36 -f bam -l 0 -S /dev/stdin
       | /opt/sambamba_0.6.3/sambamba_v0.6.3 sort -t 36 --natural-sort -m 15GiB --tmpdir ./
       -o ${if (inputs.reads != null) {return inputs.reads.nameroot} else {return ""}}.unsorted.bam -l 5 /dev/stdin
 inputs:
-  ref: { type: File, secondaryFiles: [.64.amb, .64.ann, .64.bwt, .64.pac, .64.sa, .64.alt, ^.dict, .amb, .ann, .bwt, .pac, .sa, .fai], doc: "Reference fasta file with associated indexes" }
+  ref: { type: File, secondaryFiles: [.64.amb, .64.ann, .64.bwt, .64.pac, .64.sa, .64.alt, ^.dict, .fai], doc: "Reference fasta file with associated indexes" }
```
Comment (migbro) [inline tools/bwa_mem_naive.cwl:39]: Try *.amb, etc? or document that when user creates index, must be .64

File: tools/sambamba_merge_anylist.cwl [hunk 2]
```diff
@@ -13,20 +13,35 @@ requirements:
...
+            return ret;
+        }
 baseCommand: []
 arguments:
   - position: 0
     shellQuote: false
     valueFrom: |-
       ${
-        var flatin = [].concat.apply([],inputs.bams);
+        var flatin = flatten(inputs.bams);
```
Comment (migbro) [inline tools/sambamba_merge_anylist.cwl:36]: Does this also work in cavatica?

File: tools/bwa_mem_naive.cwl (hunk 1 above)
Comment (mark-welsh) [reply tools/bwa_mem_naive.cwl:39]: @dmiller15 did you say this does not work?

File: tools/sambamba_merge_anylist.cwl (hunk 2 above)
Comment (dmiller15) [reply tools/sambamba_merge_anylist.cwl:36]: Yes

File: tools/bwa_mem_naive.cwl (hunk 1 above)
Comment (dmiller15) [reply tools/bwa_mem_naive.cwl:39]: Tested *.amb and it does not work. Will have to put it in the documentation that only .64 bwa indexes are accepted.

**Pull request merged**